bugzoo/mgr/container.py

```diff
@@ -50,6 +50,16 @@ def clear(self) -> None:
         Closes all running containers.
         """
         logger.debug("clearing all running containers")
+        try:
+            all_uids = [uid for uid in self.__containers.keys()]
+            for uid in all_uids:
+                try:
+                    self.__delitem__(uid)
+                except KeyError:
+                    # Already deleted
+                    pass
+        except AttributeError:
+            logger.debug("containers do not exist yet.")
```

Comment:
When does this happen?

Reply:
> this happens when a process has sent a `DELETE` request separately

A request to delete an individual container?

> and the container is removed after we get the list of `uid`s

If this is to account for a single container being removed after the list of UIDs is obtained, shouldn't this `except` block be indented? I don't think we should stop killing the remaining containers if one of them has already been killed.

Reply:
Oh sorry. I thought you're asking about the other except. This one is there because `clear` is called from `__init__`  and `self.__container` may not exist yet.